src/Tools/BuildValidator/CompilationDiff.cs

```diff
@@ -87,58 +144,118 @@ public static CompilationDiff CreatePlaceholder(FileInfo originalBinaryPath, boo
                 {
-                    logger.LogError($"Rebuild of {originalBinaryPath.Name} was not equivalent to the original.");
-                    if (!options.Debug)
+                    if (_localReferenceResolver.TryGetCachedAssemblyInfo(info.Mvid, out var assemblyInfo))
                     {
-                        logger.LogInformation("Pass the --debug argument and re-run to write the visualization of the original and rebuild to disk.");
+                        writer.WriteLine($"Found: {info.Mvid} {info.Name} at {assemblyInfo.FilePath}");
                     }
                     else
                     {
-                        logger.LogInformation("Creating a diff...");
-
-                        var debugPath = options.DebugPath;
-                        logger.LogInformation($@"Writing diffs to ""{Path.GetFullPath(debugPath)}""");
-
-                        fixed (byte* ptr = rebuildBytes)
+                        writer.WriteLine($"Missing: {info.Mvid} {info.Name}");
+                        foreach (var cachedInfo in _localReferenceResolver.GetCachedAssemblyInfos(info.Name))
                         {
-                            using var rebuildPeReader = new PEReader(ptr, rebuildBytes.Length);
-                            var originalInfo = new BuildInfo(
-                                AssemblyBytes: originalBytes,
-                                AssemblyReader: optionsReader.PeReader,
-                                PdbMetadataReader: optionsReader.PdbReader);
-
-                            var rebuildInfo = new BuildInfo(
-                                AssemblyBytes: rebuildBytes,
-                                AssemblyReader: rebuildPeReader,
-                                PdbMetadataReader: rebuildPeReader.GetEmbeddedPdbMetadataReader());
-
-                            createDiffArtifacts(debugPath, originalBinaryPath.Name, originalInfo, rebuildInfo, producedCompilation);
-                            SearchForKnownIssues(logger, originalInfo, rebuildInfo);
+                            writer.WriteLine($"\t{cachedInfo.Mvid} {cachedInfo.FilePath}");
                         }
                     }
                 }
+            }
 
-                return new CompilationDiff(originalBinaryPath.FullName, bytesEqual);
+            void writeBinaryDiffArtifacts()
+            {
+                Debug.Assert(Outcome == CompilationOutcome.BinaryDifference);
+                Debug.Assert(_originalPortableExecutableBytes is object);
+                Debug.Assert(_rebuildPortableExecutableBytes is object);
+                Debug.Assert(_rebuildCompilation is object);
+
+                fixed (byte* ptr = _rebuildPortableExecutableBytes)
+                {
+                    using var originalPeReader = new PEReader(ptr, _originalPortableExecutableBytes.Length);
+                    using var rebuildPeReader = new PEReader(ptr, _rebuildPortableExecutableBytes.Length);
```

Comment:
The two readers are reading from the same `ptr` buffer?

Reply:
Well that would explain the one head scratcher I was looking at 😦